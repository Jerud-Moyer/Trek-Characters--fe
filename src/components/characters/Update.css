.form {
  margin: 0;
  display: flex;
  flex-direction: column;
  margin: 20px;
  background-color: #0099f6;
  max-height: 90vh;
  overflow: scroll;

  label {
    margin-top: 30px;
    margin-bottom: 10px;
    color: #df0000
  }

  input {
    max-width: 500px;
  }
  button {
    margin-top: 30px;
    width: 100px;
    height: 30px;
    color: #df0000;
    background-color: #f2c300;
  }
}
