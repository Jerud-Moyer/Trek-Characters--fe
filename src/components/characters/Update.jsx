import React from 'react';
import PropTypes from 'prop-types';

const Update = ({
  name,
  affiliation,
  origin,
  race,
  imageUrl,
  onChange,
  onSubmit
}) => (
  <>
    <form onSubmit={onSubmit}>
      <label htmlFor="name">Name</label>
      <input
        id="name"
        type="text"
        name="name"
        value={name}
        onChange={onChange}
        
      />
      <label htmlFor="affiliation">Affiliation</label>
      <input
        id="affiliation"
        type="text"
        name="affiliation"
        value={affiliation}
        onChange={onChange}
      />
      <label htmlFor="origin">Origin</label>
      <input
        id="origin"
        type="text"
        name="origin"
        value={origin}
        onChange={onChange}
      />
      <label htmlFor="race">Race</label>
      <input
        id="race"
        type="text"
        name="race"
        value={race}
        onChange={onChange}
      />
      <label htmlFor="imageUrl">Image</label>
      <input
        id="imageUrl"
        type="url"
        name="imageUrl"
        value={imageUrl}
        onChange={onChange}
      />
      <button>Update</button>
    </form>

  </>
);

Update.propTypes = {
  name: PropTypes.string.isRequired,
  affiliation: PropTypes.string.isRequired,
  origin: PropTypes.string.isRequired,
  race: PropTypes.string.isRequired,
  imageUrl: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired
};

export default Update;
