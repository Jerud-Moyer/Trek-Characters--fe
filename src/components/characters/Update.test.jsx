import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Update from './Update';

describe('Update component', () => {
  afterEach(() => cleanup());

  it('renders Character', () => {
    const { asFragment } = render(<Update
      name="test Name"
      affiliation="Milky way defense force"
      origin="Mars"
      race="Martian"
      imageUrl="Test Image"
      onChange={() => {}}
      onSubmit={() => {}}
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
