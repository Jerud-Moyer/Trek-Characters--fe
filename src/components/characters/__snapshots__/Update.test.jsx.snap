// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Update component renders Character 1`] = `
<DocumentFragment>
  <form
    class="form"
  >
    <label
      for="name"
    >
      Name
    </label>
    <input
      id="name"
      name="name"
      type="text"
      value="test Name"
    />
    <label
      for="affiliation"
    >
      Affiliation
    </label>
    <input
      id="affiliation"
      name="affiliation"
      type="text"
      value="Milky way defense force"
    />
    <label
      for="origin"
    >
      Origin
    </label>
    <input
      id="origin"
      name="origin"
      type="text"
      value="Mars"
    />
    <label
      for="race"
    >
      Race
    </label>
    <input
      id="race"
      name="race"
      type="text"
      value="Martian"
    />
    <label
      for="imageUrl"
    >
      Image
    </label>
    <input
      id="imageUrl"
      name="imageUrl"
      type="url"
      value="Test Image"
    />
    <button>
      Update
    </button>
  </form>
</DocumentFragment>
`;
